src/test/components/Root-history.test.js

```diff
@@ -74,10 +74,20 @@ describe('Root with history', function() {
       _cleanup = null;
     };
 
-    const { findByRole } = renderResult;
-
-    async function waitForTab(details: *): Promise<HTMLElement> {
-      return findByRole('tab', details);
+    const { findByText } = renderResult;
+
+    async function waitForTab({
+      name,
+      selected,
+    }: {|
+      +name: string,
+      +selected: boolean,
+    |}): Promise<HTMLElement> {
+      // This uses `findByText` instead of `findbyRole` because this is a lot
+      // faster in our use case where there's a lot of DOM nodes.
+      return findByText(name, {
+        selector: `button[role~=tab][aria-selected=${String(selected)}]`,
```

Comment:
This won't work for implicit roles, but for tabs this works fine. I used an explicit `button` in the selector because my wild guess is jsdom optimizes for tag names (but I haven't checked), at least browsers do this.